client/options.go

```diff
@@ -0,0 +1,23 @@
+package client
+
+import (
+	"net/url"
+
+	"github.com/SKF/go-rest-utility/client/auth"
+)
+
+type Option func(*Client)
+
+func WithBaseURL(baseURL string) Option {
+	return func(c *Client) {
+		// If the provided URL is not valid the final URL will not be valid and
+		// therefore it is safe to ignore this error.
+		c.BaseURL, _ = url.Parse(baseURL) //nolint:errcheck
+	}
+}
+
+func WithTokenProvider(provider auth.TokenProvider) Option {
```

Comment:
If we copy this interface we have no dependency on go-utility? 

It is still nice to show it in the example but it feels nice to not have the dependency in the actual code.

Reply:
auth.TokenProvider is our own interface (auth is the inner auth package in the client package)

Reply:
Read this wrongly. When I realized my mistake I did not find this comment again :)